src/ui/Templates/CoreHelpers.ts

```diff
@@ -381,17 +455,34 @@ TemplateHelpers.registerTemplateHelper(
     phrasesToHighlight = resolvePhrasesToHighlight(),
     opts?: IStreamHighlightOptions
   ) => {
-    if (Utils.exists(content) && Utils.exists(termsToHighlight) && Utils.exists(phrasesToHighlight)) {
-      if (termsToHighlightAreDefined(termsToHighlight, phrasesToHighlight)) {
-        return StreamHighlightUtils.highlightStreamText(content, termsToHighlight, phrasesToHighlight, opts);
-      } else {
-        return content;
-      }
+    return executeHighlightStreamText(content, {
+      termsToHighlight,
+      phrasesToHighlight,
+      opts
+    });
+  }
+);
+
+TemplateHelpers.registerTemplateHelper('highlightStreamTextv2', (content: string, options: IHelperStreamHighlightOptions) => {
+  const mergedOptions = {
+    termsToHighlight: resolveTermsToHighlight(),
+    phrasesToHighlight: resolvePhrasesToHighlight(),
+    ...options
+  };
+  return executeHighlightStreamText(content, mergedOptions);
+});
+
+const executeHighlightStreamHTML = (content: string, options: IHelperStreamHighlightOptions) => {
+  if (Utils.exists(content) && Utils.exists(options.termsToHighlight) && Utils.exists(options.phrasesToHighlight)) {
```

Comment:
For highlighting, is it required to specify both `termsToHighlight` and `phrasesToHighlight` or just one or the other? Same comment for line 439.

Reply:
Yes, both should be specified.